artifactory/commands/transferfiles/errorshandler.go

```diff
@@ -317,3 +322,35 @@ func getErrorsFiles(repoKey string, isRetry bool) (filesPaths []string, err erro
 	}
 	return
 }
+
+// ErrorsChannelMng managing writing 'files uploading errors' to a common channel.
+// In case that an error occurred while handling the files - stops adding elements to the channel.
+type ErrorsChannelMng struct {
+	channel chan FileUploadStatusResponse
+	err     error
+}
+
+// Check if a new element can be added to the channel
```

Comment:
This description does not accurately describe this function, because this function does more than just check,